web_src/js/features/user-auth-webauthn.js

```diff
@@ -0,0 +1,243 @@
+    output += tripletToBase64(temp);
+  }
+
+  // pad the end with zeros, but make sure to not forget the extra bytes
+  switch (extraBytes) {
+    case 1:
+      temp = uint8[uint8.length - 1];
+      output += encode(temp >> 2);
+      output += encode((temp << 4) & 0x3F);
+      output += '==';
+      break;
+    case 2:
+      temp = (uint8[uint8.length - 2] << 8) + (uint8[uint8.length - 1]);
+      output += encode(temp >> 10);
+      output += encode((temp >> 4) & 0x3F);
+      output += encode((temp << 2) & 0x3F);
+      output += '=';
+      break;
+    default:
+      break;
+  }
+
+  return output;
+}
+
+// Encode an ArrayBuffer into a base64 string.
+function bufferEncode(value) {
+  return uint8ToBase64(value)
+    .replace(/\+/g, '-')
+    .replace(/\//g, '_')
+    .replace(/=/g, '');
+}
+
+function webauthnRegistered(newCredential) {
+  if (checkError(newCredential)) {
+    return;
+  }
+
+  const attestationObject = new Uint8Array(newCredential.response.attestationObject);
+  const clientDataJSON = new Uint8Array(newCredential.response.clientDataJSON);
+  const rawId = new Uint8Array(newCredential.rawId);
+
+  $.ajax({
+    url: `${appSubUrl}/user/settings/security/webauthn/register`,
+    type: 'POST',
+    headers: {'X-Csrf-Token': csrfToken},
+    data: JSON.stringify({
+      id: newCredential.id,
+      rawId: bufferEncode(rawId),
+      type: newCredential.type,
+      response: {
+        attestationObject: bufferEncode(attestationObject),
+        clientDataJSON: bufferEncode(clientDataJSON),
+      },
+    }),
+    dataType: 'json',
+    contentType: 'application/json; charset=utf-8',
+    success() {
+      window.location.reload();
+    },
+    fail() {
+      webAuthnError(1);
+    }
+  });
+}
+
+function checkError(resp) {
+  if (!('errorCode' in resp)) {
+    return false;
+  }
+  if (resp.errorCode === 0) {
+    return false;
+  }
+  webAuthnError(resp.errorCode);
+  return true;
+}
+
+function webAuthnError(errorType) {
+  const errors = {
+    browser: $('#unsupported-browser'),
+    1: $('#u2f-error-1'),
+    2: $('#u2f-error-2'),
+    3: $('#u2f-error-3'),
+    4: $('#u2f-error-4'),
+    5: $('.u2f_error_5')
+  };
+  errors[errorType].removeClass('hide');
+
+  for (const type of Object.keys(errors)) {
+    if (type !== `${errorType}`) {
+      errors[type].addClass('hide');
+    }
+  }
+  $('#u2f-error').modal('show');
+}
+
+export function initUserAuthWebAuthnRegister() {
+  detectWebAuthnSupport();
+  $('#register-device').modal({allowMultiple: false});
+  $('#u2f-error').modal({allowMultiple: false});
+  $('#register-security-key').on('click', (e) => {
+    e.preventDefault();
+    webAuthnRegisterRequest();
+  });
+}
+
+function detectWebAuthnSupport() {
+  if (window.PublicKeyCredential === undefined ||
+      typeof window.PublicKeyCredential !== 'function') {
+    $('#register-button').attr('disabled', true);
+    $('#login-button').attr('disabled', true);
```

Comment:
Using attr is incorrect.

```suggestion
    $('#register-button').prop('disabled', true);
    $('#login-button').prop('disabled', true);
```